pkg/reconciler/reconciler.go

```diff
@@ -531,6 +544,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 		)
 		return ctrl.Result{}, err
 	}
+	if state == statePending {
+		return r.handlePending(actionClient, rel, &u, log)
```

Comment:
@joelanford To be honest, I don't get to a good solution here and would like that you take a decision here.

**1. handlePending in actionClient**
My suggestion would be moving the `handlePending` to the `actionClient`. 
This makes sense as it is a Helm state which is handled such the actionClient abstracts Helm interactions.
To include `handlePending` in `handleReconcile` the `statePending` must be checked at a different execution time (within the `switch state` in line 559).

**2. Moving handlePending to the `switch state`**
As far as I see the only disadvantage is that `pre-hooks` are executed before the pending state is resolved. This is not necessarily bad and depends more on our definition of `pre-hooks/extensions` functions.